src/AbstractOperations/binary_operations.jl

```diff
@@ -187,6 +193,40 @@ macro binary(ops...)
     return expr
 end
 
+#####
+##### BinaryOperations with ZeroField
+#####
+
+import Base: +, -, *, /, ==
+using Oceananigans.Fields: ZeroField, ConstantField
+
+==(::ZeroField, ::ZeroField) = true
+
+==(zf::ZeroField, cf::ConstantField) = 0 == cf.constant
+==(cf::ConstantField, zf::ZeroField) = ==(zf, cf)
+==(c1::ConstantField, c2::ConstantField) = c1.constant == c2.constant
+
++(a::ZeroField, b::AbstractField) = b
++(a::AbstractField, b::ZeroField) = +(b, a)
++(a::ZeroField, b::Number) = ConstantField(b)
++(a::Number, b::ZeroField) = +(b, a)
+
+-(a::ZeroField, b::AbstractField) = -b
+-(b::AbstractField, a::ZeroField) = b
+-(a::ZeroField, b::Number) = ConstantField(-b)
+-(b::Number, a::ZeroField) = ConstantField(b)
```

Comment:
```suggestion
-(a::ZeroField, b::Number) = -b
-(b::Number, a::ZeroField) = b
```

is this better? It avoids the wrapper which could simplify operations (eg they can be reduced by julia lowering)

Reply:
but in this case, wouldn't we lose the information? (the result is not an abstract field anymore). I guess it depends on the end goal.

Reply:
In general I think it's likely that operations between a `ZeroField` and a `Number` are going to be used with other `AbstractFields`. So my guess is that outputting `ConstantField`s in those cases is probably advantageous more often than not.

Reply:
Okay, I think it makes sense to keep the `ConstantField` wrapper so that things remain in Field-world. Because, in fact, operations are simplified by these rules eg

```julia
julia> two = ConstantField(2)
ConstantField(2)

julia> c = CenterField(grid)
4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU
├── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
├── boundary conditions: FieldBoundaryConditions
│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: ZeroFlux, top: ZeroFlux, immersed: ZeroFlux
└── data: 10×10×10 OffsetArray(::Array{Float64, 3}, -2:7, -2:7, -2:7) with eltype Float64 with indices -2:7×-2:7×-2:7
    └── max=0.0, min=0.0, mean=0.0

julia> 3 * two * c
BinaryOperation at (Center, Center, Center)
├── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
└── tree:
    * at (Center, Center, Center)
    ├── 6
    └── 4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU
```

is the same as

```julia
julia> 3 * 2 * c
BinaryOperation at (Center, Center, Center)
├── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
└── tree:
    * at (Center, Center, Center)
    ├── 6
    └── 4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU
```

Reply:
(I was worried that `3 * two * c` would be more complicated than `3 * 2 * c`)